projects/plugins/social/src/js/components/social-notes-toggle/index.tsx

```diff
@@ -81,15 +98,17 @@ const SocialNotesToggle: React.FC< SocialNotesToggleProps > = ( { disabled } ) =
 					<ToggleControl
 						label={ __( 'Append post link', 'jetpack-social' ) }
 						checked={ appendLink }
+						disabled={ isAppendLinkToggleUpdating || isLinkFormatUpdating }
 						className={ styles.toggle }
 						onChange={ onToggleAppendLink }
 						help={ __( 'Whether to append the post link when sharing a note.', 'jetpack-social' ) }
 					/>
-					{ appendLink ? (
+					{ appendLink && ! isAppendLinkToggleUpdating ? (
```

Comment:
I think we should only disable when updating, not hide.

Reply:
I added this to the child toggle to handle these two scenarios:
- ON -> OFF : We might as well hide it at the start of the action, as they won't be able to interact with it anyway after the promise finishes
- OFF -> ON: We only show the child items when the promise is finished, as that's the first time they can interact with them.

https://github.com/Automattic/jetpack/assets/36671565/6137ad4a-f3e6-4078-a446-005a3181a8a2

What do you think? I'm also happy to change it, I just followed the pattern how we do it with the SIG settings.

Reply:
Ah OK, that makes sense.